src/Chirp.Web/Pages/UserTimeline.cshtml

```diff
@@ -10,42 +9,5 @@
 
 <div>
     <h2> @routeName's Timeline </h2>
-    @if (User.Identity.IsAuthenticated)
-    {
-        <div class="cheepbox">
-            <h3>What's on your mind, @(User.Identity.Name)?</h3>
-            <form method="post">
-                <input style="float: left" asp-for="Text">
-                <input type="submit">
-            </form>
-        </div>
-    }
-    @if (Model.Cheeps.Any())
-    {
-        <ul id="messagelist" class="cheeps">
-            @foreach (var cheep in Model.Cheeps)
-            {
-                <li>
-                    <p>
-                        <strong>
-                            <a href="/@cheep.Author.Name">@cheep.Author.Name</a>
-                        </strong>
-                        @cheep.Text
-                        <small>&mdash; @cheep.GetSerializedTimeStamp()</small>
-                    </p>
-                </li>
-            }
-        </ul>
-    }
-    else
-    {
-        <em>There are no cheeps so far.</em>
-    }
-
-    <div class="pagination">
-        @for (int i = 1; i <= Math.Ceiling((double) Model.NumOfCheeps / Model.CheepsPerPage); i++)
-        {
-            <a href="/@routeName/?page=@i">@i</a>
-        }
-    </div>
+    @await Html.PartialAsync("Shared/_CheepList")
```

Comment:
This looks pretty neat would love if you could explain it.

Reply:
This works kind of like Inserting a React component. The contents of the _CheepList file (which is something called a partial view https://learn.microsoft.com/en-us/aspnet/core/mvc/views/partial?view=aspnetcore-7.0) are inserted into the UserTimeline and inherits the model used (TimelineModel).